redisinsight/api/src/modules/database-analysis/database-analysis.service.ts

```diff
@@ -55,19 +54,28 @@ export class DatabaseAnalysisService {
         progress.total += nodeResult.progress.total;
       });
 
-      const recommendations = DatabaseAnalysisService.getRecommendationsSummary(
-        flatten(await Promise.all(
-          scanResults.map(async (nodeResult, idx) => (
-            await this.recommendationService.getRecommendations({
-              client: nodeResult.client,
-              keys: nodeResult.keys,
-              total: progress.total,
-              // TODO: create generic solution to exclude recommendations
-              exclude: idx !== 0 ? [RECOMMENDATION_NAMES.RTS] : [],
-            })
-          )),
-        )),
-      );
+      let recommendationToExclude = [];
+
+      const recommendations = await scanResults.reduce(async (previousPromise, nodeResult, idx) => {
+        const jobsArray = await previousPromise;
+        const nodeRecommendations = await this.recommendationService.getRecommendations({
+          client: nodeResult.client,
+          keys: nodeResult.keys,
+          total: progress.total,
+          globalClient: client,
+          exclude: recommendationToExclude,
+        });
+        if (idx === 0) {
+          recommendationToExclude = concat(recommendationToExclude, ONE_NODE_RECOMMENDATIONS);
+        }
+        const foundedRecommendations = nodeRecommendations.filter((recommendation) => !isNull(recommendation));
+        const foundedRecommendationNames = foundedRecommendations.map(({ name }) => name);
+        recommendationToExclude = concat(recommendationToExclude, foundedRecommendationNames);
+        recommendationToExclude.push(...foundedRecommendationNames);
+        jobsArray.push(foundedRecommendations);
+        return flatten(jobsArray);
+      }, Promise.resolve([]));
```

Comment:
In general I understand this idea. It is not readable a little bit but should work. I would prefer to have Map with recommendations when each recommendation has own type and based on it we will check this recommendation or not. But let's revise this approach and refactor current solution when we will add batch or scan more functionality